### PR TITLE
:seedling: Update `config/` so `make bundle` can update `bundle/`

### LINKED ISSUE #201: `config/` and `Makefile` bundle variables are out of date
It appears that this repo has been modifying `bundle/` without modifying `config/`

Updates should be put into config/ first, and then run `make bundle` to generate an update `bundle/`

For example [see here](https://github.com/openshift/oadp-operator/pull/860/files#diff-6d4c4e29ec096149781536bb9118fde8242c1901df764e3eda3019ac8503d6c7L152)
where we are updating config/rbac/role.yaml

The changes are [propagated to bundle/](https://github.com/openshift/oadp-operator/pull/860/files#diff-069ea2f824de6770c7836f0e18755a37580805128fe91bc747c41152efcea852) for you.

image

## TL;DR
please do not make changes to `bundle/` by hand. Make them in `config/` then run `make bundle`

#### Tips: 
clusterserviceversion in `bundle/` generated by `make bundle` pull values from both `config/` and `Makefile`

For example, bundle channels, operator image tags (ie. podman pull URL) are pulled from Makefile.
https://github.com/konveyor/tackle2-operator/blob/e72591abb03b784d00beeef86c0ef19ff0637e36/Makefile#L22-L42

You will note that utilizing config/ like example above should make editing clusterserviceversion easier for you since you can just edit role.yaml without worrying about its indentations in relations to the final clusterserviceversion yaml.

### PR DESCRIPTION
**Pull request by kaovilai**
👻 simplified tools/tackle-opdev.sh with `make bundle`
:bug: `docker-build`, `bundle-build` fixes for docker where `--arch` is invalid

:seedling: `make deploy-olm` to run local unpublished changes from source installed with a new bundle.
- replicate behavior of installing from OperatorHub giving you better idea of what the CSV, CRDs, RBACs, from the `bundle/` looks like in the cluster rather than just creating a simple deployment or replacing the deployment image.
- TL;DR `make podman-build podman-push bundle bundle-build bundle-push` using ttl.sh registry (no auth required)

Goal: minimize functional changes and allow for `make bundle` command to work using info from `config/`

```sh
diff <(curl -s https://raw.githubusercontent.com/konveyor/tackle2-operator/main/bundle/manifests/konveyor-operator.clusterserviceversion.yaml | yq -P 'sort_keys(..)') <(curl -s https://raw.githubusercontent.com/kaovilai/tackle2-operator/configmakefile-update/bundle/manifests/konveyor-operator.clusterserviceversion.yaml | yq -P 'sort_keys(..)')
```

minimal semantic diff when running through https://yamldiff.com/
- `curl https://raw.githubusercontent.com/konveyor/tackle2-operator/main/bundle/manifests/konveyor-operator.clusterserviceversion.yaml | yq -P 'sort_keys(..)'`
- `curl https://raw.githubusercontent.com/kaovilai/tackle2-operator/configmakefile-update/bundle/manifests/konveyor-operator.clusterserviceversion.yaml | yq -P 'sort_keys(..)'`

noop `bundle/` changes
- side-effect of operator-sdk generate bundle
    - yaml keys are now sorted from a to z
    - long string values are broken up into multiple lines
    - `>+` `|+` no longer has `+`

other `bundle/` changes
- :seedling: CSV name change: `tackle-operator.v99.0.0` --> `konveyor-operator.v99.0.0`
    - operator-sdk requires CSV and file names to match the [projectName](https://github.com/konveyor/tackle2-operator/blob/0c18e4fdde3de1ad148dcad9d9eb4ab269ba5b03/PROJECT#L7)
        - alternative is to change projectName to tackle-operator
    - Partial requirement of #198
- relatedImages in CSV is now generated from manager Env `RELATED_IMAGE_<>` with prefix removed, lowercased:
  - new relatedImages:
      - `oauth-proxy` previously not in CSV.
  - renamed relatedImages:
      - `tackle-addon` --> `addon-admin`
      - `tackle-addon-windup` --> `addon-windup`
      - `tackle-pathfinder` --> `pathfinder`
      - `tackle-keycloak` --> `keycloak-sso`
      - `tackle-keycloak-init` --> `keycloak-init`
- :seedling: renamed manager Env:
    - `RELATED_IMAGE_TACKLE_POSTGRES` replaces
        -  `RELATED_IMAGE_KEYCLOAK_DATABASE`
        - `RELATED_IMAGE_PATHFINDER_DATABASE`
    - solves
        ```
        WARN[0000] warning: multiple related images with the same image ref, "pathfinder-database", and different names found.The image will only be listed once with an empty name.It is recmmended to either remove the duplicate or use the exact same name. 
        ```

While this is in draft, you may see some deletions in this PR which are a result of outdated config/ and will be resolved before coming out of draft..

Closes #201 

Signed-off-by: Tiger Kaovilai <tkaovila@redhat.com>

### CODE REVIEW

**Comment by jwmatthews** [comment]
If this ends up being the direction for metadata updates, let's be sure that the info captured in https://github.com/konveyor/tackle2-operator/pull/200 is not lost.

**Comment by kaovilai** [comment]
rebased with community email update.

ready for review: @jwmatthews @jmontleon @djzager

**Comment by jmontleon** [comment]
@fbladilo @rayfordj do you foresee any headache downstream generating the csv in this way. I can't really see why it would matter, but probably worth checking.

**Comment by rayfordj** [comment]
> @fbladilo @rayfordj do you foresee any headache downstream generating the csv in this way. I can't really see why it would matter, but probably worth checking.

@jmontleon , I don't see anything immediately concerning with the proposed changes.  We should be business as usual downstream after a few initial adjustments to account for some of the things being introduced (Tackle -> Konveyor, RELATED_IMAGE, relatedImages, etc.).

In regards to potential downstream headache, I would like @fbladilo to review to ensure I'm not overlooking something he may catch, but I don't think we need to block merging if it's otherwise ready.

Thanks!

File: config/rbac/hub_service_account.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: tackle-hub
+  namespace: system
```
[the remark this answers is not in the document]
Comment (kaovilai) [reply config/rbac/hub_service_account.yaml:5]: Most things in config will have its namespace overwritten by 
https://github.com/konveyor/tackle2-operator/pull/216/files#diff-c006b1ef995eeeb57408cce42bd84db8dfc0cbe18df03222857665cebb1ac6fdR2

[the remark this answers is not in the document]
Comment (kaovilai) [reply config/rbac/hub_service_account.yaml:5]: `rm -r bundle/` and `make bundle`
and see for yourself.

[the remark this answers is not in the document]
Comment (jmontleon) [reply config/rbac/hub_service_account.yaml:5]: If this isn't used for generating the bundle should we get rid of it then? Does it have some use?

[the remark this answers is not in the document]
Comment (kaovilai) [reply config/rbac/hub_service_account.yaml]: Done.
https://github.com/konveyor/tackle2-operator/pull/216/commits/f8d1654e15988f4d585463072d3909321bfad880

[the remark this answers is not in the document]
Comment (kaovilai) [reply config/rbac/hub_service_account.yaml]: > Does it have some use?

I guess it's just a placeholder.

**Comment by fbladilo** [comment]
@kaovilai @jmontleon I'm not sure how operator is being built and published currently but we might want to update the Dockerfile version to match the operator-sdk used to test this (assuming is being used), right now is v1.22.0 which is quite dated : 

https://github.com/konveyor/tackle2-operator/blob/main/Dockerfile

At some point , a full operatort-sdk re-scaffolding of the repo might be appropiate , last time it was v1.22 .

**Comment by kaovilai** [comment]
@fbladilo no need to rescaffold.. just follow each migration guide [here](https://sdk.operatorframework.io/docs/upgrading-sdk-version/) specific to ansible operator. [43313b1](https://github.com/konveyor/tackle2-operator/commit/43313b187f7bcf353613fdd862d4b4175e7afb16)

**Review by djzager — APPROVED** [review]

**Pull request merged**